### PR TITLE
new(tests): EOF - EIP-7620: tests for msg.depth and static flag

### PR DESCRIPTION
**Pull request by pdobacz**
Mainly inspired by https://github.com/ipsilon/eof/pull/138 and evmone state_transition tests. Following the conventions used in #730, in particular the test `test_eof_eofcreate_msg_depth` is an extension of the `test_eof_calls_msg_depth`, just even more complex and magic. I'm not sure if this isn't too much, WDYT? It would however be good to have this (exotic) condition covered (and we decided we do want this `msg.depth` rule applied for various reasons).

### CODE REVIEW

**Review by marioevz — COMMENTED** [review]
A couple of comments but overall looks good.
`test_eof_eofcreate_msg_depth` is a bit hard to follow because of the memory offsets.

File: tests/prague/eip7692_eof_v1/eip7620_eof_create/test_eofcreate_failures.py [hunk 1]
```diff
@@ -585,3 +585,189 @@ def test_insufficient_returncontract_auxdata_gas(
...
+                + Op.ISZERO
+                + Op.RJUMPI[rjump_offset]
+                + returndatacopy_block
+                + deep_most_result_block
+            ),
+            Section.Container(
+                Container.Code(
+                    Op.MSTORE(0, Op.ADD(Op.CALLDATALOAD(0), 1))
+                    + Op.MSTORE(128, magic_value_call)
+                    + opcode(address=callee_address, args_size=64)
```
[the remark this answers is not in the document]
Comment (pdobacz) [reply tests/prague/eip7692_eof_v1/eip7620_eof_create/test_eofcreate_failures.py:714]: This isn't so simple, as this is initcode, so `Op.ADDRESS` is different at everydepth, also such in-deployment contract cannot call itself... I guess, I'm a bit :exploding_head: with this... Maybe let's keep it simple here and just have a static `callee_address`?

[the remark this answers is not in the document]
Comment (marioevz) [reply tests/prague/eip7692_eof_v1/eip7620_eof_create/test_eofcreate_failures.py:714]: Yes I understand, let's just add the `@pytest.mark.pre_alloc_modify` marker to this test in order to not try to run it in live networks (it's impossible simply because of the amount of gas consumed anyway).

You might need to rebase to `main` to get the marker.

**Review by marioevz — APPROVED** [review]
LGTM, please just apply the last comment in the remaining open discussion before merging.

**Comment by pdobacz** [comment]
> LGTM, please just apply the last comment in the remaining open discussion before merging.

Done, I've also noticed I should have simplified the memory layout, and then documented it a bit more (I agree it was very hard to follow)
